apps/ycrv/utils/actions/vLyCRVDeposit.ts

```diff
@@ -0,0 +1,24 @@
+import {ethers} from 'ethers';
+import {VLYCRV_TOKEN_ADDRESS} from '@yearn-finance/web-lib/utils/constants';
+
+import VLYCRV_ABI from '../abi/vlYCrv.abi';
+
+import type {TTransactionProps} from './types';
+
+export async function vLyCRVDeposit({provider, amount}: TTransactionProps): Promise<boolean> {
+	const signer = provider.getSigner();
+
+	try {
+		const contract = new ethers.Contract(VLYCRV_TOKEN_ADDRESS, VLYCRV_ABI, signer);
+		const transaction = await contract.deposit(amount);
+		const transactionResult = await transaction.wait();
+		if (transactionResult.status === 0) {
+			throw new Error('Fail to perform transaction');
+		}
+
+		return true;
+	} catch(error) {
+		console.error(error);
+		return false;
+	}
```

Comment:
Oh nice idea, extracted with https://github.com/yearn/yearn.fi/pull/49/commits/83401a6f358a60028a7042943d2cf10e76be6895

We could perhaps consolidate this handler by using it for the file you've mentioned above too

Reply:
I just don't want this PR to become massive, but happy to do it here if this PR is still manageable :)

Let me know, otherwise we could have a PR in the future to deal with this "technical debt"